catalog/ui/search-ui/search-endpoint/src/test/java/org/codice/ddf/ui/searchui/query/controller/SearchControllerTest.java

```diff
@@ -108,6 +136,42 @@ public void setUp() throws Exception {
         mockServerSession = mock(ServerSession.class);
 
         when(mockServerSession.getId()).thenReturn(MOCK_SESSION_ID);
+        when(mockSearchRequest.getId()).thenReturn("any_id");
+        when(mockSearchRequest.getSourceIds()).thenReturn(new HashSet<>(Arrays.asList(SOURCE_ID_LIST)));
+        when(mockSearchRequest.getQuery()).thenReturn(mockQuery);
+        when(mockQuery.getSortBy()).thenReturn(SortBy.NATURAL_ORDER);
+
+        when(mockBayeuxServer.getChannel(anyString())).thenReturn(mockServerChannel);
+        when(mockExecutor.submit(any(Runnable.class))).thenReturn(mockFuture);
+    }
+
+    @Test
+    public void testExceptionPassedOnBadQuery() throws Exception {
+        runTestForReasonMessages(new ExecutionException(new UnsupportedQueryException()),
+                Search.REASON_UNSUPPORTED_QUERY);
+    }
+
+    @Test
+    public void testExceptionPassedOnSourceError() throws Exception {
+        runTestForReasonMessages(new ExecutionException(new SourceUnavailableException()),
+                Search.REASON_SOURCE_UNAVAILABLE);
+    }
+
+    @Test
+    public void testExceptionPassedOnInterrupt() throws Exception {
+        runTestForReasonMessages(new InterruptedException(), Search.REASON_INTERRUPTED);
+    }
+
+    @Test
+    public void testExceptionPassedOnTimeout() throws Exception {
+        runTestForReasonMessages(new TimeoutException(), Search.REASON_TIMEOUT);
+    }
+
+    @Test
+    public void testExceptionPassedOnExecutionException() throws Exception {
+        ExecutionException executionException = new ExecutionException(
+                new ExecutionException(new Exception()));
+        runTestForReasonMessages(executionException, Search.REASON_INTERNAL);
```

Comment:
Since we added a depth limit, we should test for that as well. Or we can remove the depth limit ;-)